math/coin_problem.cpp

```diff
@@ -0,0 +1,83 @@
+/**
+ * @file
+ * @brief A program to solve the coin problem (https://en.wikipedia.org/wiki/Coin_problem)
+ * @details
+ * The coin problem is a math riddle also known as the Frobenius coin problem,
+ * postage stamp problem, and Chicken McNugget Theorem. The problem asks that
+ * given 2 items with different positive weights/value (for example coins/stamps with monetary value),
+ * what is the largest total weight/value that cannot be achieved using any combination of
+ * those two items? If the two items (a & b) are relatively prime, this can be solved using the
+ * formula (a*b)-(a+b). If they are not relatively prime, there is no solution.
+ * @author [Jason Baron](https://github.com/pkyu)
+ */
+
+#include <cassert>   /// for assert
+#include <iostream>  /// for std::cin and std::cout
+
+ /**
+  * @namespace math
+  * @brief Mathematical algorithms
+  */
+namespace math {
+
+    /**
+     * @brief Determine if the numbers are relatively prime.
+     * If they are, return the solution using Frobenius number formula.
+     * If not, there is no solution. Return -1.
+     * @param a,b input numbers
+     * @return answer
+     */
+
+    int coin_problem(int a, int b) {
+        /// Compute solution if they were relatively prime
+        int solution = (a * b) - (a + b);
+        /// Use Euclid's algorithm to find GCD
+        while (a != b) {
+            if (a > b) a = a - b;
+            else b = b - a;
+        }
+        /// If GCD is 1, a & b are relatively prime - return solution.
+        /// If GCD is not 1, there is no solution - return -1.
+        if (a == 1) return solution;
+        return -1;
+    }
+} // namespace math
+
+/**
+ * @brief Self-test implementations
+ * @returns Frobenius number
```

Comment:
This is a `void` function, therefore it doesn't return anything. :)
```suggestion
 * @returns void
```